major_tom/major_tom.py

```diff
@@ -240,3 +241,89 @@ def build_endpoints(self):
                 "files": files
             }
         })
+
+    def download_staged_file(self, gateway_download_path):
+        if self.https:
+            download_url = "https://"
+        else:
+            download_url = "http://"
+        download_url = download_url + self.host + gateway_download_path
+        r = requests.get(download_url,headers=self.headers)
+        for field in r.headers:
+            logger.debug(f'{field}  :  {r.headers[field]}')
+        if r.status_code != 200:
+            raise(RuntimeError(f"File Download Failed. Status code: {r.status_code}"))
+        filename = re.findall('filename="(.+)";', r.headers['Content-Disposition'])[0]
+        logger.info(f"Downloaded Staged File: {filename}")
+        return filename, r.content
+
+    def upload_downlinked_file(self, filename: str, filepath: str, system: str, timestamp = time.time()*1000, content_type = "binary/octet-stream", command_id = None, metadata = None):
+
+        # Get size and checksum
+        byte_size = int(os.path.getsize(filepath))
+        with open(filepath, 'rb') as file_handle:
+            checksum = b64encode(hashlib.md5(file_handle.read()).digest())
+
+        # Data to request upload
+        request_data = {
+            "filename":filename,
+            "byte_size":byte_size,
+            "content_type":content_type,
+            "checksum":checksum
+        }
+
+        # POST file info and get upload info
+        if self.https:
+            request_url = "https://"
+        else:
+            request_url = "http://"
+        request_url += self.host + "/rails/active_storage/direct_uploads"
+        logging.debug(f"Requesting {request_url} with data: {request_data}")
+        request_r = requests.post(url=request_url,headers=self.headers,data=request_data)
+        if request_r.status_code != 200:
+            logger.error(f"Transaction Failed. Status code: {request_r.status_code} \n Text Response: {request_r.text}")
+            raise(RuntimeError(f"File Upload Request Failed. Status code: {request_r.status_code}"))
+        request_content = json.loads(request_r.content)
+        for field in request_content:
+            logger.debug(f'{field}  :  {request_content[field]}')
+
+        # PUT file to MT
```

Comment:
This is putting the file to S3 or minio, not MT.

Reply:
Well yes...but for practical purposes people using this will think of it as Major Tom, since it's part of Major Tom. I don't comment above that I'm getting the upload link from Rails